src/cmd/tools/read_data_files/main/main.go

```diff
@@ -150,12 +150,20 @@ func main() {
 			start          = time.Now()
 		)
 
+		volumeNums, err := getVolumeNumber(*optPathPrefix, *optNamespace, int(shard), strconv.FormatInt(*optBlockstart, 10))
+		if err != nil {
+			log.Errorf("Failed to get volume number from file names for shard %d, using default shard number 0", shard)
+			volumeNums = []int{int(*volume)}
+		}
+
+		for volume := range volumeNums {
```

Comment:
I think you shall only try to read from the largest (a.k.a latest version), not all of them

Reply:
Does only the largest one exist? My impression is so.

Reply:
What do you mean by "largest", the largest volume number of the volume with most file counts?

Reply:
The largest volume number, like `-1-` will exist while `-0-` will be deleted